src/polysolve/nonlinear/Solver.cpp

```diff
@@ -217,7 +222,9 @@ namespace polysolve::nonlinear
                 break;
             }
 
-            this->m_current.fDelta = std::abs(old_energy - energy);
+            f_delta = std::abs(old_energy - energy);
+            // stop based on f_delta only if the solver has taken over f_delta_step_tol steps with small f_delta
+            this->m_current.fDelta = (f_delta_step_cnt > f_delta_step_tol) ? f_delta : NaN;
```

Comment:
>=? also I thought we wanted to accumulate the fdelta, not check only after n interations

Reply:
I'm thinking of `f_delta` in consecutive n steps below the tolerance, so if you want the accumulation of `f_delta` below the tolerance, you can multiply your tolerance by `n`?

Reply:
idk what is the behavior we want. right now, it is checked only after n iterations. we can sum all f deltas for 100 iters and stop if below f_delta_step_tol